src/input.rs

```diff
@@ -94,7 +94,7 @@ pub(crate) struct InputMetadata {
 pub struct Input<'a> {
     pub(crate) kind: InputKind<'a>,
     pub(crate) metadata: InputMetadata,
-    pub(crate) description: InputDescription,
+    pub description: InputDescription,
```

Comment:
I'm not a big fan of changing the visibility of this. It wasn't meant to be directly accessible through the public API, and increasing the surface area of the API will make it harder for us to change implementation details in the future without them becoming breaking changes.

You can already use `my_input.description()` to get a `&InputDescription`, anyway :)